src/codegen/array_boundary.rs

```diff
@@ -0,0 +1,102 @@
+use super::vartable::Vartable;
+use super::Options;
+use crate::ast;
+use crate::codegen::cfg::{ControlFlowGraph, Instr};
+use crate::codegen::expression::expression;
+use crate::codegen::Expression;
+use crate::sema::ast::{Namespace, Type};
+use indexmap::IndexMap;
+use num_bigint::BigInt;
+use num_traits::One;
+use solang_parser::pt::Loc;
+
+// IndexMap <ArrayVariable res , res of temp variable>
+pub type ArrayLengthVars = IndexMap<usize, usize>;
+
+// Function to modify arraye length temp
+pub(crate) fn modify_temp_array_length(
```

Comment:
What is your opinion about making this a method of `ControlFlowGraph`?
@seanyoung

Reply:
I agree it would be a better design, but was just leaving it till the end